src/main/kotlin/ar/edu/unahur/obj2/caralibro/Publicacion.kt

```diff
@@ -3,14 +3,93 @@ package ar.edu.unahur.obj2.caralibro
 import kotlin.math.ceil
 
 abstract class Publicacion {
+  var usuariosQueLeGusta = mutableListOf<Usuario>()
   abstract fun espacioQueOcupa(): Int
+
+  abstract fun puedeSerVistaPorUnUsuario(usuarioQueQuiereVerla: Usuario, usuarioQueLaTiene: Usuario) : Boolean
+
+  //cambie a la forma que dijiste vos
+  fun cantidadDeMeGustasQueTieneLaPublicacion() = usuariosQueLeGusta.size
+
+  fun agregarUnUsuarioAUsuariosQueLeDieronMeGusta(usuarioQueLeDioMg: Usuario){
+    check(!usuariosQueLeGusta.contains(usuarioQueLeDioMg)){
+      "El usuario ya le dio me gusta a la publicacion"
+    }
+    usuariosQueLeGusta.add(usuarioQueLeDioMg)
+  }
```

Comment:
Si usaban un `MutableSet` se ahorraban el tener que chequear si estaba repetido.
```suggestion
val usuariosQueLeGusta = mutableSetOf<Usuario>()
  fun agregarUnUsuarioAUsuariosQueLeDieronMeGusta(usuarioQueLeDioMg: Usuario){
      usuariosQueLeGusta.add(usuarioQueLeDioMg)
  }
```

Reply:
Está perfecto. Me parece que nos quedamos con el requerimiento del ejercicio anterior que tenía que avisar cuando era repetido. En este caso no era necesario. Gracias!!!